### DISCUSSION TITLE
GLobalMax pool for dense GCN

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#2592] [Q&A]
How can I apply global max-pool for DenseGCn?
because if I apply global max-pool on the output of DenseGCn then an error occurs of mismatched dimension as global max-pool layer wants the batch size.

**Comment by rusty1s** [comment]
In dense-mode, you can simply use `x.max(dim=1)[0]` for global max pooling, where `x` has shape `[batch_size, num_nodes, num_features]`.

**Reply by shrimonmuke0202** [reply]
RuntimeError: batch1 dim 2 must match batch2 dim 1
this error occurs for applying DenseGCN on the adjacency matrix.

**Reply by rusty1s** [reply]
Can you show me an example on what you are trying to do and what fails?

**Reply by shrimonmuke0202** [reply]
x , edge_index, batch = data.x,data.edge_index,data.batch
    adj = to_dense_adj(edge_index,batch)
    print(adj.shape)
    target = data.target
    
    if self.k1 == 1:
      print("1")
      h1 = self.Conv1(x,adj)

**Reply by rusty1s** [reply]
You need to convert the `x` to a dense representation as well via `x, _ = to_dense_batch(x, batch)`

**Reply by shrimonmuke0202** [reply]
Thank you
